hclwrite/parser_test.go

```diff
@@ -555,6 +555,78 @@ func TestParse(t *testing.T) {
 				},
 			},
 		},
+		{
+			"a = foo[bar]\n",
```

Comment:
What `hclwrite`'s "parser" is doing here is turning the raw token stream into a physical parse tree, as opposed to the _abstract_ parse tree in `hclsyntax`.

It uses the source ranges from `hclsyntax` to find the bounds of the attribute definition and its expression, and then it further marks out the traversals inside the expression in order to enable "rename symbol"-type operations in text editors to surgically update the references without affecting the surrounding tokens.

This case was panicking before because the `foo` token was not being indicated by `hclsyntax` as part of the expression, and so when the traversal analyzer attempted to find the `TokenIdent` to learn the name it was looking into an empty token sequence.